### PR TITLE
[Fix] `jsx-no-leaked-render`: autofix nested "&&" logical expressions

### LINKED ISSUE #3323: Bug: `jsx-no-leaked-render` faulty autofix
Version: 7.30.1
Config:
```js
'react/jsx-no-leaked-render': ['error', { validStrategies: ['coerce', 'ternary'] }],
```

Before fix:
```tsx
const MyComponent = () => {
  const items = []
  const breakpoint = { phones: true }

  return <div>{items.length > 0 && breakpoint.phones && <span />}</div>
}
```

Post fix:

```tsx
const MyComponent = () => {
  const items = []
  const breakpoint = { phones: true }

  return <div>{!!!!!!!!!!!!!!!!!!!!items.length > 0 && breakpoint.phones && <span />}</div>
}
```

Expected:

```tsx
const MyComponent = () => {
  const items = []
  const breakpoint = { phones: true }

  return <div>{items.length > 0 && !!breakpoint.phones && <span />}</div>
}
```

On top of that, the auto fix is quite slow.

### PR DESCRIPTION
**Pull request by hduprat**
A partial fix to #3323

I welcome suggestions and help if you think it does not improve performance 🙂

### CODE REVIEW

**Review by Belco90 — CHANGES_REQUESTED** [review]
LGTM! Could you add the scenarios reported in #3323 as valid tests? So it's clear that those cases have been fixed.

File: lib/rules/jsx-no-leaked-render.js [hunk 1]
```diff
@@ -40,19 +40,28 @@ function getIsCoerceValidNestedLogicalExpression(node) {
   return COERCE_VALID_LEFT_SIDE_EXPRESSIONS.some((validExpression) => validExpression === node.type);
 }
 
+function extractExpressionBetweenLogicalAnds(node) {
+  if (node.type !== 'LogicalExpression') return [node];
+  if (node.operator !== '&&') return [node];
+  const op = [];
+  return op.concat(extractExpressionBetweenLogicalAnds(node.left), extractExpressionBetweenLogicalAnds(node.right));
```
Comment (ljharb) [inline lib/rules/jsx-no-leaked-render.js:47]:
```suggestion
  return [].concat(
    extractExpressionBetweenLogicalAnds(node.left),
    extractExpressionBetweenLogicalAnds(node.right)
  );
```

**Review by ljharb — CHANGES_REQUESTED** [review]
Looks great, but could we add the test case from the OP of the linked issue?

File: lib/rules/jsx-no-leaked-render.js (hunk 1 above)
Comment (hduprat) [reply lib/rules/jsx-no-leaked-render.js]: @ljharb it's done now

**Review by Belco90 — APPROVED** [review]

**Review by ljharb — APPROVED** [review]

**Pull request merged**